### PR TITLE
Fix MvcRenderViewToString and add a test to verify it works

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @dougbu \ @ajaybhargavb

File: samples/Mvc.RenderViewToString/Program.cs [hunk 1]
```diff
@@ -33,25 +42,38 @@ public static void Main()
...
+                WebRootFileProvider = fileProvider,
             });
             services.Configure<RazorViewEngineOptions>(options =>
             {
                 options.FileProviders.Clear();
-                options.FileProviders.Add(new PhysicalFileProvider(customApplicationBasePath ?? applicationEnvironment.ApplicationBasePath));
+                options.FileProviders.Add(fileProvider);
             });
             var diagnosticSource = new DiagnosticListener("Microsoft.AspNetCore");
+            services.AddSingleton<ObjectPoolProvider, DefaultObjectPoolProvider>();
```
Comment (pranavkm) [inline samples/Mvc.RenderViewToString/Program.cs]: @davidfowl - this is so utterly crummy.

File: samples/Mvc.RenderViewToString/project.json [hunk 2]
```diff
@@ -1,7 +1,10 @@
 {
   "buildOptions": {
     "emitEntryPoint": true,
-    "preserveCompilationContext": true
+    "preserveCompilationContext": true,
+    "copyToOutput": {
+      "include": "Views"
```
Comment (pranavkm) [inline samples/Mvc.RenderViewToString/project.json]: We don't set the content root to the current working directory anywhere so this is required to make `dotnet run` work  out of the box.

File: test/FunctionalTests/MvcTests/RazorEmbeddedResource.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Reflection;
+using Xunit;
+
+namespace EntropyTests
+{
+    public static class RazorEmbeddedResource
```
Comment (dougbu) [inline test/FunctionalTests/MvcTests/RazorEmbeddedResource.cs]: Any changes from the `TagHelperSampleTest` source?

File: samples/Mvc.RenderViewToString/Program.cs (hunk 1 above)
Comment (dougbu) [inline samples/Mvc.RenderViewToString/Program.cs]: How would you prefer hosting-provided services be handled?

**Comment by dougbu** [comment]
LGTM but I defer to @ajaybhargavb on the squirrel.

File: test/FunctionalTests/MvcTests/RazorEmbeddedResource.cs (hunk 3 above)
Comment (pranavkm) [inline test/FunctionalTests/MvcTests/RazorEmbeddedResource.cs]: The method was made public and it's referencing this type to locate the project, but nothing apart from that.

File: samples/Mvc.RenderViewToString/Program.cs (hunk 1 above)
Comment (pranavkm) [inline samples/Mvc.RenderViewToString/Program.cs]: Since Razor depends on it and we sort of expect it to work in non-hosted scenarios, we should add it as part of `AddRazor` \ `AddMvc`. It's a no-op operation if it already exists via Hosting.

**Comment by pranavkm** [comment]
Ping @ajaybhargavb

**Comment by ajaybhargavb** [comment]
:shipit:

**Comment by ajaybhargavb** [comment]
Does it make sense to add one more test with `customApplicationBasePath=null`?

**Comment by pranavkm** [comment]
It's kinda tricky because you have to in some way modify the `ApplicationBasePath` to point to the test app when it's pointing at the test project. The end results are going to look pretty similar to passing in a path.

**Comment by ajaybhargavb** [comment]
Got it
